confidant/routes/v1.py

```diff
@@ -374,6 +383,10 @@ def get_credential(id):
             'Item with id {0} does not exist.'.format(id)
         )
         return jsonify({}), 404
+
+    if not authnz.require_saml_role(cred.metadata)['read_only']:
```

Comment:
I think it's maybe better to have this as a decorator that can be applied to endpoints, rather than being checked in-function.

For example:

```
@authnz.require_role('read_only')
```

Reply:
I will work on this